lib/languages/js/templates/api.js.tpl

```diff
@@ -133,54 +119,48 @@ module.exports = function (options) {
       }
     }
     req.url = (opts.url || '{{{options.root}}}') + req.url;
-    req.paramsSerializer = function(params) { return qs.stringify(params); };
-    var promise = axios(req, cb)
-      .then(function (response) {
-        response = response.data;
-        if (cb) { return setTimeout(function () { cb(null, response); }, 0); }
-        return response;
-      })
-      .catch(function (axiosError) {
-        var err;
+    req.paramsSerializer = (params) => { return qs.stringify(params); };
+    const reqPromise = async () => {
+      try {
+        const response = await axios(req);
+        return response.data;
+      } catch (axiosError) {
+        let err;
         if (axiosError.response) {
-          var errorData = axiosError.response.data || {};
+          const errorData = axiosError.response.data || {};
           err = new Error(errorData.message);
-          Object.keys(errorData).forEach(function(key) {
+          Object.keys(errorData).forEach((key) => {
             if (key !== 'message') { err[key] = errorData[key]; }
           });
           err.statusCode = axiosError.response.status;
           if (err.statusCode === 413 && !err.message) { err.message = 'Request entity too large.'; }
         } else {
           err = axiosError;
         }
-        if (cb) { return setTimeout(function () { cb(err); }, 0); }
         throw err;
-      });
-    if (!cb) { return promise; }
+      }
+    };
+
+    return reqPromise();
```

Comment:
why bother making this a function and calling it? just make request an async function?